mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -1127,31 +1129,45 @@ public Future<Void> deleteRecords(int lastUpdatedDays, int limit, String tenantI
    */
   @Override
   public Future<Boolean> deleteMarcIndexersOldVersions(String tenantId) {
+    return executeInTransaction(txQE -> acquireLock(txQE, tenantId)
+      .compose(isLockAcquired -> {
+        if (Boolean.FALSE.equals(isLockAcquired)) {
+          LOG.info("deleteMarcIndexersOldVersions:: Previous marc_indexers old version deletion still ongoing, tenantId: '{}'", tenantId);
+          return Future.succeededFuture(false);
+        }
+        return deleteMarcIndexersOldVersions(txQE, tenantId);
+      }), tenantId);
+  }
+
+  private Future<Boolean> acquireLock(ReactiveClassicGenericQueryExecutor txQE, String tenantId) {
+    return txQE.findOneRow(dsl -> dsl.select(DSL.field("{0}", Boolean.class,
+      DSL.function(GET_LOCK_FUNCTION, SQLDataType.BOOLEAN, val(tenantId.hashCode()))))
```

Comment:
Thinking maybe we should have a hashcode on a prefix like "delete_marc_indexers_"+tenantId instead of just tenant id. Just so to avoid collisions with actors that may just use the tenant id as the lock id.

Reply:
added prefix usage for lock obtaining, just initially was thinking of adding prefix/additional lock id on demand when the same approach will be used